src/codeFlowLogin.ts

```diff
@@ -340,27 +342,32 @@ async function startServer(server: http.Server, adfs: boolean) {
 	return port;
 }
 
-/* eslint-disable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access */
 async function callback(nonce: string, reqUrl: url.Url): Promise<string> {
-	let error = reqUrl.query.error_description || reqUrl.query.error;
-
-	if (!error) {
-		const state = reqUrl.query.state || '';
-		// eslint-disable-next-line @typescript-eslint/no-unsafe-call
-		const receivedNonce = (state.split(',')[1] || '').replace(/ /g, '+');
-		if (receivedNonce !== nonce) {
-			error = 'Nonce does not match.';
+	let query: ParsedUrlQuery;
+	let error: string | undefined;
+	let code: string | undefined;
+
+	if (reqUrl.query) {
+		query = typeof reqUrl.query === 'string' ? parse(reqUrl.query) : reqUrl.query;
+		error = query.error_description.toString() || query.error.toString() || '';
```

Comment:
I believe the properties on `query` will be `undefined` if they're not specified in the url, so by adding `toString()` you could be causing a null ref error. There's a lot of ways to fix that, but I'd probably end up writing a little helper function
```typescript
function getQueryProp(query: ParsedUrlQuery, propName: string): string {
    const value = query[propName];
    return typeof value === 'string' ? value : '';
}
```
```typescript
error = getQueryProp(query, 'error_description') || getQueryProp(query, 'error');
```